src/frontend/packages/semantic-data-provider/src/dataProvider/dataProvider.js

```diff
@@ -66,7 +67,10 @@ const dataProvider = config => {
       await fetchUserConfigPromise;
       await fetchVoidEndpointsPromise;
       return config;
-    }
+    },
+    uploadFile: waitForConfig(async rawFile => {
+      return await handleFiles.uploadFile(rawFile, config);
+    })
```

Comment:
You can do without the async/await if you return immediately an await function:
```suggestion
    uploadFile: waitForConfig(rawFile => uploadFile(rawFile, config))
```